ios/MeasureSDKTests/Exporter/BatchCreatorTests.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  BatchCreatorTests.swift
+//  MeasureSDKTests
+//
+//  Created by Adwin Ross on 19/10/24.
+//
+
+@testable import MeasureSDK
+import XCTest
+
+final class BatchCreatorTests: XCTestCase {
+    var batchCreator: BatchCreator!
+    var logger: MockLogger!
+    var idProvider: MockIdProvider!
+    var configProvider: MockConfigProvider!
+    var timeProvider: MockTimeProvider!
+    var eventStore: MockEventStore!
+    var batchStore: MockBatchStore!
+
+    override func setUp() {
+        super.setUp()
+        logger = MockLogger()
+        idProvider = MockIdProvider()
+        configProvider = MockConfigProvider()
+        timeProvider = MockTimeProvider()
+        eventStore = MockEventStore()
+        batchStore = MockBatchStore()
+
+        batchCreator = BaseBatchCreator(
+            logger: logger,
+            idProvider: idProvider,
+            configProvider: configProvider,
+            timeProvider: timeProvider,
+            eventStore: eventStore,
+            batchStore: batchStore
+        )
+    }
+
+    override func tearDown() {
+        batchCreator = nil
+        logger = nil
+        idProvider = nil
+        configProvider = nil
+        timeProvider = nil
+        eventStore = nil
+        batchStore = nil
+        super.tearDown()
+    }
+
+    func testCreateWithNoEvents() {
+        eventStore.events = []
+
+        let result = batchCreator.create(sessionId: nil)
+
+        XCTAssertNil(result)
+        let batches = batchStore.getBatches(5)
+        XCTAssertTrue(batches.isEmpty)
+    }
+
+    func testCreateWithEventsButExceedsAttachmentSize() {
+        eventStore.addEvents()
+        configProvider.maxAttachmentSizeInEventsBatchInBytes = 100
+        configProvider.maxEventsInBatch = 2
+
+        let result = batchCreator.create(sessionId: nil)
+        let batches = batchStore.getBatches(5)
+
+        XCTAssertNil(result)
+        XCTAssertTrue(batches.isEmpty)
+    }
+
+    func testCreateSuccessfulBatch() {
+        eventStore.addEvents()
```

Comment:
Create events for each test within the test function to keep test localized. Moving event creation to a separate class would also help.